### PR TITLE
Fix crash when removing popup menu item after signal index_pressed called

### PR DESCRIPTION
**Pull request by khairul169**
This pr will fix issue #19842

### CODE REVIEW

File: scene/gui/popup_menu.cpp [hunk 1]
```diff
@@ -1019,7 +1019,7 @@ void PopupMenu::activate_item(int p_item) {
 	//hide all parent PopupMenue's
 	Node *next = get_parent();
 	PopupMenu *pop = Object::cast_to<PopupMenu>(next);
-	while (pop) {
+	while (pop && p_item >= 0 && p_item < items.size()) {
```
Comment (vnen) [inline scene/gui/popup_menu.cpp]: That's odd, there's already an `ERR_FAIL_INDEX` at the beginning, so it should never reach this if `p_item` is not within the `items.size()` boundary.

**Comment by khairul169** [comment]
That's caused by both signals are called after p_item check

**Comment by arthurferrai** [comment]
I think the issue main problem is not wether the index exists but why it pass a non-existing index to signal as a parameter.
In the issue example, the second signal should send index 0 instead of 1.

**Comment by akien-mga** [comment]
Superseded by #20982. Thanks for your contribution!

**Pull request closed without merging**